components/layout/dom_traversal.rs

```diff
@@ -75,7 +81,15 @@ impl<'dom> NodeAndStyleInfo<'dom> {
     }
 
     pub(crate) fn get_selected_style(&self) -> ServoArc<ComputedValues> {
-        self.node.to_threadsafe().selected_style()
+        if self.node.is_single_line_text_inner_editor() {
```

Comment:
I'm still a bit skeptical about this part...

What is `get_selected_style` used for? Can we avoid this hack by properly inheriting the relevant properties from the shadow host?

Reply:
The `get_selected_style` is being used in the display list to draw the selection. it would be a little bit awkward, since we need to get the styles from `input::selection`.

Reply:
Does not seems to be the case in Gecko though. There are an open issue for this https://bugzilla.mozilla.org/show_bug.cgi?id=1749166.

Reply:
Discussed offline as this part is still necessary to allow drawing *any* selection for an input. So this part is preserved but a comment is added.